File: AppNumber/app/src/main/java/com/devheverton/appnumber/MainActivity.java
package com.devheverton.appnumber;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;

import com.devheverton.appnumber.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        binding.btCalcular.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                calcular();
            }
        });
    }
    private void calcular(){
        int numero = Integer.parseInt(binding.editNumero.getText().toString());

        if ( numero % 2 == 0){
            binding.txtResultado.setText("O número "  +  numero  + " é Par");
        }else{
            binding.txtResultado.setText("O número " +  numero  + " é Impar");
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_principal, menu);
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        int itemID = item.getItemId();

        if (itemID == R.id.ic_limap){
            binding.editNumero.setText("");
            binding.txtResultado.setText("");
        }
        return super.onOptionsItemSelected(item);
    }
}

File: GeradordeSenha/app/src/main/java/com/devheverton/geradordesenha/MainActivity.kt
package com.devheverton.geradordesenha

import android.os.Bundle

import androidx.appcompat.app.AppCompatActivity
import com.devheverton.geradordesenha.databinding.ActivityMainBinding
import kotlin.random.Random
import kotlin.random.nextInt


class MainActivity : AppCompatActivity() {

    private  lateinit var binding: ActivityMainBinding
    private val alfabeto = arrayOf("A", "B","C","D","E","F","G","H","I","J","K","L","M","N","O","P","Q","R","S","T","U","V","W","X","Y","Z")

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

     binding.btGerarSenha.setOnClickListener {

         val letrAleatoria1 = alfabeto.random()
         val letrAleatoria2 = alfabeto.random()
         val numero = Random.nextInt(1000..9999)

         binding.txtSenha.text = "$letrAleatoria1$letrAleatoria2-$numero"


     }

        }



    }
